evalml/tests/component_tests/test_target_imputer.py

```diff
@@ -0,0 +1,185 @@
+
+    y = pd.Series([np.nan, 1, 1, 2])
+    imputer = TargetImputer(impute_strategy='most_frequent')
+    y_expected = pd.Series([1, 1, 1, 2])
+    _, y_t = imputer.fit_transform(None, y)
+    assert_series_equal(y_expected, y_t.to_series(), check_dtype=False)
+
+
+def test_target_imputer_col_with_non_numeric_with_numeric_strategy():
+    y = pd.Series([np.nan, "a", "b"])
+    imputer = TargetImputer(impute_strategy='mean')
+    with pytest.raises(ValueError, match="Cannot use mean strategy with non-numeric data"):
+        imputer.fit_transform(None, y)
+    with pytest.raises(ValueError, match="Cannot use mean strategy with non-numeric data"):
+        imputer.fit(None, y)
+    imputer = TargetImputer(impute_strategy='median')
+    with pytest.raises(ValueError, match="Cannot use median strategy with non-numeric data"):
+        imputer.fit_transform(None, y)
+    with pytest.raises(ValueError, match="Cannot use median strategy with non-numeric data"):
+        imputer.fit(None, y)
+
+
+@pytest.mark.parametrize("data_type", ['pd', 'ww'])
+def test_target_imputer_all_bool_return_original(data_type, make_data_type):
+    y = pd.Series([True, True, False, True, True], dtype=bool)
+    y = make_data_type(data_type, y)
+    y_expected = pd.Series([True, True, False, True, True], dtype='boolean')
+    imputer = TargetImputer()
+    imputer.fit(None, y)
+    _, y_t = imputer.transform(None, y)
+    assert_series_equal(y_expected, y_t.to_series())
+
+
+@pytest.mark.parametrize("data_type", ['pd', 'ww'])
+def test_target_imputer_boolean_dtype(data_type, make_data_type):
+    y = pd.Series([True, np.nan, False, np.nan, True], dtype='boolean')
+    y_expected = pd.Series([True, True, False, True, True], dtype='boolean')
+    y = make_data_type(data_type, y)
+    imputer = TargetImputer()
+    imputer.fit(None, y)
+    _, y_t = imputer.transform(None, y)
+    assert_series_equal(y_expected, y_t.to_series())
+
+
+def test_target_imputer_fit_transform_all_nan_empty():
+    y = pd.Series([np.nan, np.nan])
+
+    imputer = TargetImputer()
+    imputer.fit(None, y)
+    with pytest.raises(RuntimeError, match="Transformed data is empty"):
+        imputer.transform(None, y)
+
+    imputer = TargetImputer()
+    with pytest.raises(RuntimeError, match="Transformed data is empty"):
+        imputer.fit_transform(None, y)
+
+
+def test_target_imputer_numpy_input():
+    y = np.array([np.nan, 0, 2])
+    imputer = TargetImputer(impute_strategy='mean')
+    y_expected = np.array([1, 0, 2])
+    _, y_t = imputer.fit_transform(None, y)
+    assert np.allclose(y_expected, y_t.to_series())
+    np.testing.assert_almost_equal(y, np.array([np.nan, 0, 2]))
+
+
+def test_target_imputer_does_not_reset_index():
+    y = pd.Series(np.arange(10))
+    y[5] = np.nan
+    assert y.index.tolist() == list(range(10))
+
+    y.drop(0, inplace=True)
+    pd.testing.assert_series_equal(pd.Series([1.0, 2, 3, 4, np.nan, 6, 7, 8, 9], dtype=float, index=list(range(1, 10))), y)
```

Comment:
Is there a reason you typed `1.0` here instead of just `1`? hyper-nit, but it seems to disrupt the flow

Reply:
Hehe nope, just copied from `test_simple_imputer_does_not_reset_index`, will update both!